### PR TITLE
Moved the metadata from setup.py to setup.cfg

### PR DESCRIPTION
**Pull request by KOLANICH**

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -1,7 +1,49 @@
...
+    pgpy.packet
+    pgpy.packet.subpackets
+install_requires = 
+    cryptography>=2.6
+    pyasn1
+    six>=1.9.0
+    singledispatch; python_version < '3.4'
+    enum34; python_version < '3.4'
+setup_requires = setuptools>=44; wheel
+# doc_requires = sphinx; sphinx-better-theme
```
Comment (J-M0) [inline setup.cfg:40]: Please reformat `setup_requires` and `doc_requires` as lists like `install_requires`

File: setup.py [hunk 2]
```diff
@@ -1,92 +1,39 @@
...
-
-    # package hierarchy
-    packages = [
-        "pgpy",
-        "pgpy.packet",
-        "pgpy.packet.subpackets"
-    ],
-)
+from pathlib import Path
+
```
Comment (J-M0) [inline setup.py:6]: Can you rewrite this file so that it will also work on Python 2.7 and 3.4? Also please use snake_case for your function names.

**Comment by J-M0** [comment]
The tests still fail when installing on 2.7 and 3.4. Please look at the test output to see the problem or edit your PR to let me push changes to it.

**Comment by KOLANICH** [comment]
>Could not find a version that satisfies the requirement setuptools>=44

It seems like setuptools has dropped 3.4. :( I wonder if we should just make the latest setuptools compatible to 3.4 again in a form of a patch/patcher or just downgrade from toml-based configuration for setuptools_scm to the one inlined into setup.py.

**Comment by J-M0** [comment]
Yeah it stinks, but we just need to do this for one more release.

From my tests, all we need to do is remove the `>=44` from `pyproject.toml` and `setup.cfg` and it will install properly after also fixing `setup.py`. If you let me push changes to your PR I can show you.

File: setup.py [hunk 3]
```diff
@@ -1,92 +1,44 @@
...
-
-    # package hierarchy
-    packages = [
-        "pgpy",
-        "pgpy.packet",
-        "pgpy.packet.subpackets"
-    ],
-)
+try:
+    from pathlib import Path
```
Comment (J-M0) [inline setup.py:7]: Please use `os.path` instead of `pathlib`

**Comment by J-M0** [comment]
I'm going to merge this in, but next time please listen if you are asked to use one particular library over another.

**Comment by KOLANICH** [comment]
IMHO: python 2 should be just dropped.

**Comment by J-M0** [comment]
I totally agree with you that python 2 should be dropped, which is why we're going to drop everything < 3.5 in PGPy 0.6.0. We just need to get 0.5.3 out the door before that happens.

Also I do want to thank you for your contributions. I'm sorry if I came off a little harsh earlier. I'm still getting up to speed with this project so please do bear with me.

**Comment by KOLANICH** [comment]
Dropping 3.4 is probably not a good idea. This is the latest version supporting Windows XP, which is still used in some orgs for the systems where upgrade is impossible or doesn't make any sense. The reason called for dropping XP was that new Visual Studio runtime (they have started building python against) doesn't support it ... but I guess the real reason was that they just desired to drop something to reduce testing and maybe remove workarounds, and since XP has been dropped by M$ by that time, they have chosen, that they will drop XP.

I wonder if it is possible to build 3.9 using MinGW-w64 (whichhas a mode supporting XP) and if it is, why isn't it done by PSF and why do they use the proprietary VS.

BTW, https://ntcore.com/?p=458 and https://github.com/Skulltrail192/One-Core-Api

**Comment by J-M0** [comment]
I see, interesting. My main concern with dropping 3.4 was to be able to start putting type hints in the codebase and I didn't realize that 3.4 supported function annotations. I will think on it some more, but considering that Windows XP has been out of support since 2014 and Python 3.4 since March 2019 I'm not sure how much we will gain by keeping 3.4 support.

**Pull request merged**